tests/e2e/v2_curl_test.go

```diff
@@ -155,6 +156,8 @@ func cURLPrefixArgs(clus *etcdProcessCluster, method string, req cURLReq) []stri
 				cmdArgs = append(cmdArgs, "--cacert", caPath, "--cert", certPath3, "--key", privateKeyPath3)
 			}
 		}
+	} else if req.useCertAuth {
```

Comment:
I am having a hard time following the logic on how the command is generated. Does the if (line 145) make sense to you? For example, I tried to test `TestV3MetricsSecure`, but the actual curl command in the test was `curl -L http://localhost:20000/metrics`

Reply:
This particular test I mentioned above was not affected by the changes in your PR. But I want to make sure the tests are correct and reader-friendly.

Reply:
> I am having a hard time following the logic on how the command is generated. 

@jingyih Me too. I had to add one more block to be able to use another certificate. It seems to me wrong that they are hardcoded in this function. On the other hand, they are pregenerated and their purpose is predefined. So maybe this makes sense.

> Does the if (line 145) make sense to you?

No, it does not make sense to me. This code seems broken. It looks like someone tried to add the logic for secure connections, but failed.

> For example, I tried to test `TestV3MetricsSecure`, but the actual curl command in the test was `curl -L http://localhost:20000/metrics`.

This is because this test does not test a secure connection :) The `TestV3MetricsSecure` and `TestV3MetricsInsecure` create custom `cfg`, but don't use it:
https://github.com/etcd-io/etcd/blob/024f3dfc820e93e912c4b031b37edc66c5a0d72c/tests/e2e/metrics_test.go#L25-L37

 To use custom config they must use `withCfg()`. It means that  these tests use the default config:
https://github.com/etcd-io/etcd/blob/024f3dfc820e93e912c4b031b37edc66c5a0d72c/tests/e2e/ctl_v3_test.go#L133-L141

As you can see there is no `metricsURLScheme` definition here:
https://github.com/etcd-io/etcd/blob/024f3dfc820e93e912c4b031b37edc66c5a0d72c/tests/e2e/cluster_test.go#L42-L47

It means the `cx.cfg.metricsURLScheme` will be empty as well:
https://github.com/etcd-io/etcd/blob/024f3dfc820e93e912c4b031b37edc66c5a0d72c/tests/e2e/metrics_test.go#L43

Reply:
Heh. If I change `req.metricsURLScheme == "https"`, then the following tests fail:
```
--- FAIL: TestV3MetricsSecureTLSCertAuth (0.58s)
--- FAIL: TestV2CurlAllTLS (0.26s)
--- FAIL: TestV2CurlClientTLS (0.96s)
--- FAIL: TestV3CurlCipherSuitesValid (0.66s)
--- FAIL: TestV3CurlPutGetAllTLS (0.91s)
--- FAIL: TestV3CurlPutGetClientTLS (1.52s)
--- FAIL: TestV3CurlAuthClientTLSCertAuth (0.81s)
--- FAIL: TestV3MetricsSecureTLSCertAuth (1.19s)
--- FAIL: TestV2CurlAllTLS (0.36s)
--- FAIL: TestV2CurlClientTLS (0.76s)
--- FAIL: TestV3CurlCipherSuitesValid (0.16s)
--- FAIL: TestV3CurlPutGetAllTLS (1.92s)
--- FAIL: TestV3CurlPutGetClientTLS (0.72s)
--- FAIL: TestV3CurlAuthClientTLSCertAuth (1.32s)
--- FAIL: TestV3MetricsSecureTLSCertAuth (0.38s)
--- FAIL: TestV2CurlAllTLS (1.06s)
--- FAIL: TestV2CurlClientTLS (0.26s)
--- FAIL: TestV3CurlCipherSuitesValid (0.66s)
--- FAIL: TestV3CurlPutGetAllTLS (1.12s)
--- FAIL: TestV3CurlPutGetClientTLS (1.22s)
--- FAIL: TestV3CurlAuthClientTLSCertAuth (1.62s)
```